protocol/activity_center_persistence.go

```diff
@@ -59,11 +60,29 @@ func (db sqlitePersistence) DeleteActivityCenterNotificationForMessage(chatID st
 		matchNotifications = append(matchNotifications, a)
 	}
 
+	var sameLastMessageNotifications []*ActivityCenterNotification
+	var nextLastMessage *common.Message
 	for _, notification := range notifications {
 		if notification.LastMessage != nil && notification.LastMessage.ID == messageID {
-			withNotification(notification)
+			sameLastMessageNotifications = append(sameLastMessageNotifications, notification)
 		}
+	}
+
+	if len(sameLastMessageNotifications) > 0 {
+		sort.Slice(sameLastMessageNotifications, func(i, j int) bool {
+			return sameLastMessageNotifications[i].Timestamp > sameLastMessageNotifications[j].Timestamp
+		})
+		nextLastMessage = sameLastMessageNotifications[0].LastMessage
+	}
 
+	var updatedNotifications []*ActivityCenterNotification
+
+	for _, notification := range notifications {
+		if notification.LastMessage != nil && notification.Message != nil && notification.Message.ID != messageID && notification.LastMessage.ID == messageID {
+			notification.LastMessage = nextLastMessage
```

Comment:
@yqrashawn, a refactor suggestion is to extract the logic to find the next last message in a private function, and then the it would be just `notification.LastMessage = findNextLastMessage(messageID, notifications)`.

Reply:
thanks, fixed